data-mongo/src/main/java/io/micronaut/data/mongo/serde/DataSerdeRegistry.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.mongo.serde;
+
+import io.micronaut.context.BeanContext;
+import io.micronaut.context.annotation.Replaces;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.beans.BeanProperty;
+import io.micronaut.core.type.Argument;
+import io.micronaut.data.document.serde.IdPropertyNamingStrategy;
+import io.micronaut.data.document.serde.IdSerializer;
+import io.micronaut.data.document.serde.ManyRelationSerializer;
+import io.micronaut.data.document.serde.OneRelationSerializer;
+import io.micronaut.data.model.runtime.AttributeConverterRegistry;
+import io.micronaut.data.model.runtime.RuntimeEntityRegistry;
+import io.micronaut.data.model.runtime.RuntimePersistentEntity;
+import io.micronaut.serde.Deserializer;
+import io.micronaut.serde.Encoder;
+import io.micronaut.serde.Serde;
+import io.micronaut.serde.SerdeIntrospections;
+import io.micronaut.serde.Serializer;
+import io.micronaut.serde.exceptions.SerdeException;
+import io.micronaut.serde.support.DefaultSerdeRegistry;
+import io.micronaut.serde.support.deserializers.ObjectDeserializer;
+import io.micronaut.serde.support.serializers.ObjectSerializer;
+import jakarta.inject.Singleton;
+import org.bson.codecs.configuration.CodecRegistry;
+
+import java.io.IOException;
+
+/**
+ * Micronaut Data serde registry.
+ *
+ * @author Denis Stepanov
+ * @since 3.3
+ */
+@Singleton
+@Replaces(DefaultSerdeRegistry.class)
+@Internal
+final class DataSerdeRegistry extends DefaultSerdeRegistry {
```

Comment:
Can we use delegation instead of inheritance here and annotate the with `@Order` with a higher priority?